### PR TITLE
feat: limit field sizes

### PR DESCRIPTION
**Pull request by d3or**

### CODE REVIEW

File: packages/indexer/src/config/index.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const config = {
   // When running in liquidity-only mode, all metadata processes are disabled
   liquidityOnly: Boolean(Number(process.env.LIQUIDITY_ONLY)),
   metadataIndexingMethod: String(process.env.METADATA_INDEXING_METHOD || "opensea"),
+  metadataMaxFieldSize: Number(process.env.METADATA_MAX_FIELD_SIZE || 1),
```
Comment (nofir) [inline packages/indexer/src/config/index.ts:24]: is this in mb? maybe we can be more explicit metadataMaxFieldSizeMB / metadataMaxFieldSizeKB

File: packages/indexer/src/metadata/providers/utils.ts [hunk 2]
```diff
@@ -182,3 +184,27 @@ export class CollectionNotFoundError extends Error {
...
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function limitFieldSize(value: any) {
+  if (typeof value === "string") {
+    logger.info(
+      "limitFieldSize",
+      JSON.stringify({
+        size: new TextEncoder().encode(value).length,
+      })
+    );
+    return new TextEncoder().encode(value).length > config.metadataMaxFieldSize * 1024 * 1024
```
Comment (nofir) [inline packages/indexer/src/metadata/providers/utils.ts:197]: are there any considerations as far as efficiency with encoding the value? i see this: https://stackoverflow.com/questions/9864662/how-to-get-the-string-length-in-bytes-in-nodejs
could be that its the same effort

File: packages/indexer/src/metadata/providers/utils.ts [hunk 3]
```diff
@@ -182,3 +184,27 @@ export class CollectionNotFoundError extends Error {
...
+      ? null
```
Comment (nofir) [inline packages/indexer/src/metadata/providers/utils.ts:198]: im wondering if we should just log in the start before but if the env var is set high so it should be ok

File: packages/indexer/src/config/index.ts (hunk 1 above)
Comment (nofir) [reply packages/indexer/src/config/index.ts:24]: or at least leave a comment

File: packages/indexer/src/metadata/providers/utils.ts (hunk 2 above)
Comment (d3or) [reply packages/indexer/src/metadata/providers/utils.ts]: i think it should be fine, both methods convert to utf8 to find the bytes, i dont expect one or the other to have a significant difference

**Review by nofir — DISMISSED** [review]

**Review by nofir — APPROVED** [review]

**Review by nofir — APPROVED** [review]

**Pull request merged**